packages/client/src/components/SafeSettings.js

```diff
@@ -266,9 +265,14 @@ const AddSafeOwner = ({ web3, onCancel, onNext }) => {
 
   const onAddressChange = async (newAddress) => {
     setAddress(newAddress);
-    setAddressValid(isAddr(newAddress) && (await isAddressValid(newAddress)));
+    setAddressValid(isAddr(newAddress) && (await isAddressValid(newAddress)) && !isAddressExisting(safeOwners, newAddress));
   };
 
+  const isAddressExisting = (safeOwners, newAddress) => {
+    const address = formatAddress(newAddress);
+    return safeOwners.filter(obj => obj.address === address).length !== 0;
```

Comment:
We should make sure the `verified` field gets evaluated as well
```suggestion
    return safeOwners.filter(obj => obj.address === address && obj.verified).length !== 0;
```

Reply:
Added!